src/engine.rs

```diff
@@ -29,20 +30,22 @@ use wgpu::{
 pub type Error = Box<dyn std::error::Error>;
 
 #[derive(Clone, Copy)]
-pub struct ShaderId(usize);
+pub struct ShaderId(pub usize);
 
 #[derive(Clone, Copy, PartialEq, Eq, Hash)]
-pub struct Id(NonZeroU64);
+pub struct Id(pub NonZeroU64);
 
 static ID_COUNTER: AtomicU64 = AtomicU64::new(0);
 
+#[cfg(feature = "wgpu")]
 pub struct Engine {
```

Comment:
These sections together are a bit strange

This suggests the engine module is too entangled with wgpu/that these IDs shouldn't be in this module.
If this is the wgpu engine implementation, it should maybe be more separated?

Reply:
It seems like at least there was an intention behind splitting `Engine` and `Recording` up, maybe moving all of the `Engine` stuff to a `wgpu.rs` file is better going forward?

Reply:
Fwiw, anybody implementing another `Engine` type thing would need access to these internals to be able to translate between `Recording` and it's own internal representation since the `Engine` is responsible for handing these out.